### PR TITLE
Button: updated styles for icon fill on hover

### LINKED ISSUE #28470: [Bug]: react-button should not fill icon on hover for primary/secondary/outline variants
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9.25.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/zen-dawn-ggm8hh

## Actual Behavior
When Button uses bundled icon, it fills icon on hover for all appearances. This is introduced in #22587.

## Expected Behavior
According to [design](https://www.figma.com/file/Nj9EBBvOZmS11zKNJfilVR/Button?node-id=4428%3A11702&mode=dev), only subtle/transparent variants should fill icon on hover. The others (primary/secondary/outline) should keep using regular icon.
image

I notice in [Button appearance story](https://react.fluentui.dev/?path=/docs/components-button-button--default#appearance), we used regular icon on primary/default variants, and bundled icon on others. Does this mean we deliver fill icon on hover for all appearances, but encourage bundled icon only for subtle/transparent?

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by kkakroo**
## Previous Behavior
Icon fill on hover was implemented on all variants of button

## New Behavior
Icon fill on hover only works only on subtle and transparent variants of button

## Related Issue(s)

- Fixes #28470

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">57</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount.html">13</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Avatar-mount_base.html">590</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Avatar-mount.html">613</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Button-mount_base.html">298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Button-mount.html">298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Field-mount_base.html">1022</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Field-mount.html">1011</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProvider-mount_base.html">657</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProvider-mount.html">645</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-mount_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-mount.html">75</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">57</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">64</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount.html">13</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/MakeStyles-mount_base.html">850</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/MakeStyles-mount.html">840</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Persona-mount_base.html">1598</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/Persona-mount.html">1546</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/SpinButton-mount_base.html">1281</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28616/perf-test-react-components/SpinButton-mount.html">1304</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `84.992 kB`<br />`22.03 kB` | `84.931 kB`<br />`22.061 kB` | `-61 B` <br />`31 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |  `39.193 kB`<br />`9.743 kB` |  `39.132 kB`<br />`9.767 kB` | `-61 B` <br />`24 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    | `46.661 kB`<br />`11.257 kB` |   `46.6 kB`<br />`11.285 kB` | `-61 B` <br />`28 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            | `43.553 kB`<br />`10.979 kB` | `43.492 kB`<br />`11.002 kB` | `-61 B` <br />`23 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          | `51.758 kB`<br />`12.571 kB` | `51.697 kB`<br />`12.597 kB` | `-61 B` <br />`26 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        | `57.629 kB`<br />`11.645 kB` | `57.568 kB`<br />`11.672 kB` | `-61 B` <br />`27 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              | `67.576 kB`<br />`18.225 kB` | `67.515 kB`<br />`18.254 kB` | `-61 B` <br />`29 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `213.111 kB`<br />`59.12 kB` | `213.05 kB`<br />`59.154 kB` | `-61 B` <br />`34 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                     |  `89.36 kB`<br />`25.284 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                               | `83.778 kB`<br />`23.685 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                   |    `9.24 kB`<br />`3.907 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                   |  `11.652 kB`<br />`4.741 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                 |  `10.048 kB`<br />`4.254 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `36.409 kB`<br />`12.003 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                              |    `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/67204719005015e0af0e272fd86d44fc02e1e0a0'>67204719005015e0af0e272fd86d44fc02e1e0a0</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-button-63f8f566-a6b7-4492-92ac-56510c1cd6df.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-button-63f8f566-a6b7-4492-92ac-56510c1cd6df.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28616&commitId=829bb7fba472c8b41cf3fb5518b213bff0768a4b&env=prod&clientType=FLUENTUI)


  <details><summary><b>Button Converged</b> 18 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button Converged.Circular and icon only - RTL.pressed.chromium.png|97| Changed|
|Button Converged.Circular and icon only.hover.chromium.png|108| Changed|
|Button Converged.Circular and icon only.pressed.chromium.png|97| Changed|
|Button Converged.Circular and icon only - RTL.hover.chromium.png|108| Changed|
|Button Converged.Icon only.pressed.chromium.png|97| Changed|
|Button Converged.Icon only.hover.chromium.png|108| Changed|
|Button Converged.Size large.hover.chromium.png|152| Changed|
|Button Converged.Size small.pressed.chromium.png|97| Changed|
|Button Converged.Size large.pressed.chromium.png|136| Changed|
|Button Converged.Size small.hover.chromium.png|108| Changed|
|Button Converged.With icon before content - RTL.pressed.chromium.png|97| Changed|
|Button Converged.With icon before content.hover.chromium.png|108| Changed|
|Button Converged.With icon after content.hover.chromium.png|108| Changed|
|Button Converged.With icon before content.pressed.chromium.png|97| Changed|
|Button Converged.With icon before content - RTL.hover.chromium.png|108| Changed|
|Button Converged.With icon after content - RTL.hover.chromium.png|108| Changed|
|Button Converged.With icon after content - RTL.pressed.chromium.png|97| Changed|
|Button Converged.With icon after content.pressed.chromium.png|97| Changed|

</details>


  <details><summary><b>CompoundButton Converged</b> 14 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|CompoundButton Converged.Circular and icon only.hover.chromium.png|443| Changed|
|CompoundButton Converged.Circular and icon only - RTL.pressed.chromium.png|398| Changed|
|CompoundButton Converged.Circular and icon only.pressed.chromium.png|398| Changed|
|CompoundButton Converged.Icon only.pressed.chromium.png|398| Changed|
|CompoundButton Converged.Circular and icon only - RTL.hover.chromium.png|443| Changed|
|CompoundButton Converged.Size large.hover.chromium.png|443| Changed|
|CompoundButton Converged.Icon only.hover.chromium.png|443| Changed|
|CompoundButton Converged.Size small.hover.chromium.png|443| Changed|
|CompoundButton Converged.Size large.pressed.chromium.png|398| Changed|
|CompoundButton Converged.Size small.pressed.chromium.png|398| Changed|
|CompoundButton Converged.With icon after content.pressed.chromium.png|398| Changed|
|CompoundButton Converged.With icon before content - RTL.pressed.chromium.png|398| Changed|
|CompoundButton Converged.With icon after content - RTL.pressed.chromium.png|398| Changed|
|CompoundButton Converged.With icon before content.pressed.chromium.png|398| Changed|

</details>


  <details><summary><b>MenuButton Converged</b> 10 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|MenuButton Converged.Circular and icon only.hover.chromium.png|108| Changed|
|MenuButton Converged.Icon only.hover.chromium.png|108| Changed|
|MenuButton Converged.Icon only.pressed.chromium.png|97| Changed|
|MenuButton Converged.Circular and icon only.pressed.chromium.png|97| Changed|
|MenuButton Converged.Size large.hover.chromium.png|152| Changed|
|MenuButton Converged.Size small.hover.chromium.png|108| Changed|
|MenuButton Converged.Size small.pressed.chromium.png|97| Changed|
|MenuButton Converged.Size large.pressed.chromium.png|136| Changed|
|MenuButton Converged.With icon.hover.chromium.png|108| Changed|
|MenuButton Converged.With icon.pressed.chromium.png|97| Changed|

</details>


  <details><summary><b>ToggleButton Converged</b> 12 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|ToggleButton Converged.Circular and icon only.hover.chromium.png|108| Changed|
|ToggleButton Converged.Circular and icon only.pressed.chromium.png|97| Changed|
|ToggleButton Converged.Icon only.hover.chromium.png|108| Changed|
|ToggleButton Converged.Icon only.pressed.chromium.png|97| Changed|
|ToggleButton Converged.Size large.pressed.chromium.png|136| Changed|
|ToggleButton Converged.Size large.hover.chromium.png|152| Changed|
|ToggleButton Converged.Size small.pressed.chromium.png|97| Changed|
|ToggleButton Converged.Size small.hover.chromium.png|108| Changed|
|ToggleButton Converged.With icon before content.pressed.chromium.png|97| Changed|
|ToggleButton Converged.With icon after content.hover.chromium.png|108| Changed|
|ToggleButton Converged.With icon after content.pressed.chromium.png|97| Changed|
|ToggleButton Converged.With icon before content.hover.chromium.png|108| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-button-63f8f566-a6b7-4492-92ac-56510c1cd6df.json (hunk 1 above)
Comment (khmakoto) [reply change/@fluentui-react-button-63f8f566-a6b7-4492-92ac-56510c1cd6df.json:1]: The screenshot changes are expected.

**Pull request merged**